.github/workflows/wsl-example.yaml

```diff
@@ -0,0 +1,88 @@
+name: Example action using WSL
+
+concurrency: azure-vm
+
+on:
+  pull_request:
+     paths:
+      # We run it when changing one if its dependencies
+      - .github/workflows/wsl-example.yaml
+      - .github/actions/**
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+env:
+  az_name: wsl-ci
+  az_resource_group: wsl
+
+jobs:
+  vm-setup:
+    # This job starts your Azure VM and opens up to RDP connections. It'll keep running
+    # in the background to host RDP connections until the machine is stopped by stop-vm.
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Start Azure VM
+        uses: ./.github/actions/vm-setup
+        with:
+          az_name: ${{ env.az_name }}
+          az_resource_group: ${{ env.az_resource_group }}
+          az_creds: ${{ secrets.AZURE_VM_CREDS }}
+          az_vm_hostname: ${{ secrets.AZURE_VM_AUTHORITY }}
+          az_vm_username: ${{ secrets.AZURE_VM_UN }}
+          az_vm_password: ${{ secrets.AZURE_VM_UP }}
+
+  run-script:
+    # This job has an implicit dependency on start-vm, because the runner
+    # won't be started until the VM is turned on. It must run concurrently with
+    # vm-setup, so don't add an explicit depenendency.
+    # Don't do this: needs: vm-setup
+    runs-on: self-hosted
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install or update WSL
+        uses: ./.github/actions/wsl-install
+        with:
+          distro: Ubuntu
+
+      - name: Checkout your repo into WSL
+        uses: ./.github/actions/wsl-checkout
+        with:
+          distro: ubuntu
```

Comment:
So, we have some jobs with the same parameter but not the same meaning if I’m correct:
- in `wsl-install`, the `distro` parameter is the appx to install (Name), hence `Ubuntu`
- in the other steps, the `distro` parameter of the actions is the wsl instance that we want to run it. I think you are using `ubuntu` here, no capital, because it’s the name the distro is registered under? Or it doesn’t matter (and in this case we should use the same case so that we don’t loose the user).

Reply:
Distro always refers to the value that appears when you do `wsl --list --verbose`.

It's windows, so not case-sensitive. You're right that it is confusing, I'll change it to always be `Ubuntu` since that is what shows up when you run the aforementioned command.